src/main/java/uk/gov/hmcts/reform/sscs/service/JudicialRefDataService.java

```diff
@@ -34,31 +31,29 @@ public String getJudicialUserFullName(String personalCode) {
         return judicialUsers.get(0).getFullName();
     }
 
-    public String getJudicialUserTitleWithInitialsAndLastName(String personalCode) {
-        String name = getJudicialUserFullName(personalCode);
-
+    public JudicialUserBase getJudicialUserFromPersonalCode(@NonNull String personalCode) {
         IdamTokens idamTokens = idamService.getIdamTokens();
 
-        JudicialRefDataSearchRequest judicialRefDataUsersRequest = JudicialRefDataSearchRequest.builder()
-            .searchString(name).build();
+        JudicialRefDataUsersRequest judicialRefDataUsersRequest = JudicialRefDataUsersRequest.builder()
+                .personalCodes(List.of(personalCode)).build();
+
+        List<JudicialUser> judicialUsers = judicialRefDataApi.getJudicialUsers(idamTokens.getIdamOauth2Token(),
+                idamTokens.getServiceAuthorization(), judicialRefDataUsersRequest);
 
-        List<JudicialUserSearch> judicialUsers = judicialRefDataApi.searchUsersBySearchString(
-            idamTokens.getIdamOauth2Token(), idamTokens.getServiceAuthorization(), judicialRefDataUsersRequest);
+        if (judicialUsers.size() > 0) {
+            JudicialUser judicialUser = judicialUsers.get(0);
 
-        for (JudicialUserSearch judicialUser : judicialUsers) {
-            if (nonNull(judicialUser.getPersonalCode()) && judicialUser.getPersonalCode().equals(personalCode)) {
-                return StringUtils.convertNameToTitleInitalsAndSurname(judicialUser);
-            }
+            return new JudicialUserBase(judicialUser.getSidamId(), judicialUser.getPersonalCode());
         }
 
-        return StringUtils.getInitalsAndSurnameFromName(name);
```

Comment:
What are the consequences of returning null here instead of the initial/surname?

Reply:
its the diff which has made it look weird, I deleted an old function that would get the name of the person and now the new function gets the judicial user object (so we can only pass null)